tests/ChannelMessagesTest.php

```diff
@@ -146,19 +148,109 @@ public function testPublishSingleMessageUnencrypted() {
     }
 
     /**
+     * Verify that batch sending messages actually makes just one request
+     */
+    public function testMessageArraySingleRequest() {
+        $messages = array();
+        
+        for ( $i = 0; $i < 10; $i++ ) {
+            $msg = new Message();
+            $msg->name = 'msg'.$i;
+            $msg->data = 'test string'.$i;
+            $messages[] = $msg;
+        }
+
+        $ably = new AblyRest( array_merge( self::$defaultOptions, array(
+            'key' => self::$testApp->getAppKeyDefault()->string,
+            'httpClass' => 'tests\HttpMockMsgCounter',
+        ) ) );
+
+        $channel = $ably->channel( 'singleReq' );
+        $channel->publish( $messages );
+
+        $this->assertEquals( 1, $ably->http->requestCount, 'Expected 1 request to be made' );
+    }
+
+    /**
+     * Verify that publishing invalid types fails
+     */
+    public function testInvalidTypes() {
+        $channel = self::$ably->channel( 'invalidTypes' );
+        
+        $msg = new Message();
+        $msg->name = 'int';
+        $msg->data = 81403;
+
+        $this->setExpectedException( 'Ably\Exceptions\AblyException', '', 40003 );
+        $channel->publish( $msg );
```

Comment:
Wouldn't this test still pass if this line wasn't throwing an exception but others did?